### PR TITLE
Reapply "[ARM] Use addLiveIns method instead of manually adding live ins (#87560)"

### PR DESCRIPTION
**Pull request by AZero13**

### CODE REVIEW

**Comment by AtariDreams** [comment]
@nikic @MaskRay thoughts on this?

**Review by nikic — COMMENTED** [review]
I'm not familiar with this code, but given the test diffs, this is clearly not NFC, right?

**Comment by AtariDreams** [comment]
> I'm not familiar with this code, but given the test diffs, this is clearly not NFC, right?

File: llvm/test/CodeGen/ARM/constant-islands-split-IT.mir
Comment (nikic) [inline llvm/test/CodeGen/ARM/constant-islands-split-IT.mir:1]: Please rebase over 00ae6bb6c27b5ca89fb22dff99a34d6b45e3e0b0.

**Comment by davemgreen** [comment]
From what I can tell from https://reviews.llvm.org/D66319 this should be OK, as it was added mostly to catch verification errors and as far as I understand not adding subregs is OK. @efriedma-quic any objections?

**Pull request closed without merging**